frontend/src/component/insights/componentsChart/CreationArchiveChart/CreationArchiveChart.tsx

```diff
@@ -51,6 +53,42 @@ interface ICreationArchiveChartProps {
     labels: { week: string; date: string }[];
 }
 
+const batchSize = 4;
+
+const sharedDatasetOptions = (lineColor: string) => ({
+    backgroundColor: lineColor,
+    borderColor: lineColor,
+    hoverBackgroundColor: lineColor,
+    hoverBorderColor: lineColor,
+});
+const makeChartData = (
+    data: FinalizedWeekData[] | BatchedWeekData[],
+    theme: Theme,
+) => ({
+    datasets: [
+        {
+            label: 'Flags archived',
+            data: data,
+            order: 1,
+            parsing: {
+                yAxisKey: 'archivedFlags',
+                xAxisKey: 'date',
+            },
+            ...sharedDatasetOptions(theme.palette.charts.A2),
+        },
+        {
+            label: 'Flags created',
+            data: data,
+            order: 2,
+            parsing: {
+                yAxisKey: 'totalCreatedFlags',
+                xAxisKey: 'date',
+            },
+            ...sharedDatasetOptions(theme.palette.charts.A1),
+        },
+    ],
+});
+
```

Comment:
These have been extracted from the `useMemo` invocation to make it easier to read and reason about.